### DISCUSSION TITLE
Connecting to the CORE Demon

### DISCUSSION THREAD

**Discussion by pjrwlazlo** [coreemu/core#591] [General]
Howdy,

I am attempting to connect with the CORE daemon through the Python scripting libraries that CORE provides. According the developers guide it the **core-pygui** can connect to session being hosted by the **core-daemon**, but the **core-pygui** is not able to find the session in my script.

It is my understanding that you can configure the network in the Python3 script, then connect to the session via the **core-pygui**.

I am using a virtual environment in for the pyCharm IDE for Python3, could this be the source of the issue?

The Python script is also running within a VM on a Type II hypervisor.

**Comment by bharnden** [comment]
You would need to run the `core-daemon` then create the session by way of gRPC to allow `core-pygui` to see the same data as they both would be talking over the gRPC API.

If your script was creating things directly within python, there would be no API running to talk to.

**Comment by pjrwlazlo** [comment]
Howdy bharnden,

That solved my issue! Thank you for pointing me in the right direction. 

For anyone having the same issue, here is a link to the CORE gRPC documentation I used to refactor my code: [https://coreemu.github.io/core/grpc.html](https://coreemu.github.io/core/grpc.html)
